Mathlib/SetTheory/Cardinal/Basic.lean

```diff
@@ -917,6 +924,18 @@ def prod {ι : Type u} (f : ι → Cardinal) : Cardinal :=
 theorem mk_pi {ι : Type u} (α : ι → Type v) : #(∀ i, α i) = prod fun i => #(α i) :=
   mk_congr <| Equiv.piCongrRight fun _ => outMkEquiv.symm
 
+theorem mk_pi_congr {ι : Type u} {f g : ι → Type v} (h : ∀ i, #(f i) = #(g i)) :
+    #(Π i, f i) = #(Π i, g i) := by
+  simp_all only [mk_pi]
+
+theorem mk_pi_congr_subtype {ι : Type u} {f g : ι → Type v} {S : Set ι}
+    (h : ∀ i ∈ S, #(f i) = #(g i)) : #(Π i ∈ S, f i) = #(Π i ∈ S, g i) :=
+  have h' : Π i ∈ S, f i ≃ g i := fun i is ↦ Classical.choice <| Cardinal.eq.mp (h i is)
+  mk_congr
+  ⟨fun f' i is ↦ h' i is (f' i is) , fun g' i is ↦ (h' i is).symm (g' i is),
+  fun _ ↦ by simp only [Equiv.symm_apply_apply],
```

Comment:
Can this `simp only` and the one below be replaced by `rw`?

Reply:
I couldn't, the thing to prove is basically `Equiv.symm_apply_apply` inside a bunch of binders and lambdas, and after intoducing them all you need to at least `simp_rw` anyways because it's not in exactly the right form.